test/infrastructure/kind/mapper.go

```diff
@@ -0,0 +1,293 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Package kind provide utilities for CAPD interoperability with kindest/node images.
+
+CAPD doesn't have direct dependencies from Kind, however, it is using kindest/node images.
+
+As a consequence CAPD must stay in sync with Kind with regards to how kindest/node images are
+run when creating a new DockerMachine.
+
+kindest/node images are created using a specific Kind version, and thus it is required
+to keep into account the mapping between the Kubernetes version for a kindest/node and
+the Kind version that created that image, so we can run it properly.
+
+NOTE: This is the same reason why in the Kind release documentation there is a specific list
+of kindest/node images - uniquely identified by a sha - to be used with a Kind release.
+
+For sake of simplification, we are grouping a set of Kind versions that runs kindest/node images
+in the same way into Kind mode.
+*/
+package kind
+
+import (
+	"fmt"
+
+	"github.com/blang/semver"
+
+	clusterapicontainer "sigs.k8s.io/cluster-api/util/container"
+)
+
+const (
+	kindestNodeImageName = "kindest/node"
+)
+
+// Mode defines a set of Kind versions that are running images in a consistent way.
+type Mode string
+
+const (
+	// ModeNone defined a Mode to be used when creating the CAPD load balancer, which doesn't
+	// use a kindest/node image.
+	ModeNone Mode = "None"
+
+	// Mode1_19 is a Mode that identifies kind v1.19 and older; all those versions
+	// are running images in a consistent way (or are too old at the time of this implementation).
+	Mode1_19 Mode = "kind 1.19"
```

Comment:
```suggestion
	Mode0_19 Mode = "kind 0.19"
```
same below and in godoc